io.go

```diff
@@ -45,6 +45,9 @@ func WriteLines(writer io.Writer) Filter {
 				return err
```

Comment:
I think we might as well get rid of the buffer and simplify.  Something like:
  if _, err := writer.Write(append([]byte(s), '\n')); err != nil { return err }
This might do an extra allocation (due to the append), but that should be fine given the file IO implicit here.